src/tunnel.cc

```diff
@@ -1015,10 +1121,11 @@ TunnelStateData::connectToPeer()
 void
 TunnelStateData::connectedToPeer(Security::EncryptorAnswer &answer)
 {
-    if (ErrorState *error = answer.error.get()) {
-        answer.error.clear(); // sendError() will own the error
-        sendError(error, "TLS peer connection error");
```

Comment:
@chtsanti , I have a question you probably may help with. Should not we call sendError() here only if clientExpectsConnectResponse() check returns true? For example, TunnelStateData::tunnelEstablishmentDone() already does this check before calling sendError() which makes sense. I assume that this is a bug and fixed it in the branch, but I would like you to confirm that my decision was correct.

Reply:
Yep, it does not make sense to send an HTTP error if the client does not "speak" HTTP.

In other places too the sendError is used without check the clientExpectsConnectResponse() before. 

I must note that the sendError also updates the TunnelStateData::*status_ptr which is used to log the status code in squid access log. This is must updated even if we do not want to send the error to the client. The TunnelStateData::tunnelEstablishmentDone() update it even if the sendError is not called.

A solution is to replace the TunnelStateData::sendError(error) calls with a TunnelStateData::sendErrorOrClose(error) which updates always the status_ptr, and checks clientExpectsConnectResponse() before send the error or close the connection.

I do not know if this is should fixed inside this PR. In any case inside TunnelStateData::retryOrBail when the sendError is not called we should update the status_ptr to the error->httpStatus.

Reply:
>  This is must updated even if we do not want to send the error to the client.

TunnelStateData::*status_ptr points to al->http.code, which is logged as ">Hs" and described as "HTTP status code sent to the client". However, in our case, we do not send the HTTP error response and, hence, the status code to the client. Something is wrong here: either the code not updating 'status_ptr' or the documentation lying about the sent error code. What do you think?

Reply:
We must update `*status_ptr` as Christos mentioned.

Related %codes documentation is indeed imprecise (e.g. `sent` really means `sent or would have been sent if ...` in most cases), but, as you know, recent attempts at improving documentation precision faced significant non-technical hurdles, so I would leave it as is.

Reply:
The only I can say is that the status code is useful to be logged even if it is not really sent to the client. In any case if not updated still we may log something which is not the final error. For example:
  - TunnelStateData::tunnelEstablishmentDone called with error.
  - TunnelStateData::tunnelEstablishmentDone updates status_ptr and call retryOrBail
  - At the next try the TunnelStateData connected OK but calls Encryptor which fails
  - The retryOrBail does not send the error to client and does not update the status_ptr 
  - Squid logs the status of the first failed transaction, which maybe is not the same as the second one

Maybe there are other similar cases too.
My opinion is that if needed we should update the  >Hs documentation. But I think squid users expects this behavior even if not documented.

Reply:
Done at a63d463.